students/rimlee_talukdar/lesson03/assignment/src/basic_operations.py

```diff
@@ -0,0 +1,157 @@
+"""
+Basic operation of customer database
+"""
+import logging
+import peewee
+
+logging.basicConfig(filename='basic_operations.log',
+                    filemode='w',
+                    format='%(asctime)s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S',
+                    level='DEBUG')
+__database__ = peewee.SqliteDatabase("customers.db")
+__database__.connect()
+__database__.execute_sql('drop table if exists customer;')
+
+class BaseModel(peewee.Model):
+    """
+    Peewee base model
+    """
+    class Meta:
+        """
+        Meta field of the database
+        """
+        database = __database__
+
+
+class Customer(BaseModel):
+    """
+    This is the customer model. Following model stores the following data
+        customer_id field for Customer ID
+        name field for Name
+        lastname field for Lastname
+        address field for Home address
+        phone_number field for Phone Number
+        email field for Email address
+        status field for Status (active or inactive customer)
+        credit_limit field for Credit Limit
+    """
+
+    customer_id = peewee.CharField(primary_key=True, max_length=30)
+    name = peewee.CharField(max_length=50)
+    lastname = peewee.CharField(max_length=50, null=True)
+    address = peewee.CharField(max_length=75)
+    phone_number = peewee.CharField(max_length=15)
+    email = peewee.CharField(max_length=320)
+    status = peewee.CharField(max_length=10)
+    credit_limit = peewee.FloatField()
+
+
+def add_customer(customer_id, name, lastname, address,
+                 phone_number, email, status, credit_limit):
+    """
+    Adds customer to the database
+    :param customer_id field for Customer ID
+    :param name field for Name
+    :param lastname field for Lastname
+    :param address field for Home address
+    :param phone_number field for Phone Number
+    :param email field for Email address
+    :param status field for Status (active or inactive customer)
+    :param credit_limit field for Credit Limit
+    :return: None
+    """
+    try:
+        with __database__.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                address=address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+            )
+            customer.save()
+            logging.info(f'Added {customer_id}/{name} to the database')
+    except Exception as exception:
```

Comment:
Similar to assignment 2, try to avoid catching the general Exception. In this case, you might look to use the peewee.DoesNotExist exception